statement_test.go

```diff
@@ -534,3 +536,60 @@ func TestBindJSON(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, string(jsonData), str)
 }
+
+func TestPrepareComplexQueryParameter(t *testing.T) {
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	ctx := context.Background()
+	createTable(t, db, `CREATE OR REPLACE TABLE arr_test(
+		arr_float float[2]
+	)`)
```

Comment:
Can we also add a test for nested lists/arrays? Something like `float[2][2]` or `varchar[][][]`.